src/main/scala/akka/persistence/cassandra/query/EventsByTagFetcher.scala

```diff
@@ -114,29 +116,8 @@ private[query] class EventsByTagFetcher(
           seqNumbers.isNext(pid, seqNr) match {
             case SequenceNumbers.Yes | SequenceNumbers.PossiblyFirst =>
               seqNumbers = seqNumbers.updated(pid, seqNr)
-              val m = row.getBytes("message") match {
-                case null =>
-                  PersistentRepr(
-                    payload = deserializeEvent(serialization, row),
-                    sequenceNr = row.getLong("sequence_nr"),
-                    persistenceId = row.getString("persistence_id"),
-                    manifest = row.getString("event_manifest"),
-                    deleted = false,
-                    sender = null,
-                    writerUuid = row.getString("writer_uuid")
-                  )
-                case b =>
-                  // for backwards compatibility
-                  persistentFromByteBuffer(b)
-              }
-
-              val eventEnvelope = UUIDEventEnvelope(
-                offset = offs,
-                persistenceId = pid,
-                sequenceNr = row.getLong("sequence_nr"),
-                event = m.payload
-              )
-              replyTo ! eventEnvelope
+              replyTo ! UUIDPersistentRepr(offs, toPersistentRepr(row))
```

Comment:
a minor optimization might be to reuse the `pid` and `seqNr`, now we read them twice from the `row`